leaves.go

```diff
@@ -0,0 +1,219 @@
+	var leaveTypes []LeaveType
+
+	resp, err := c.get(leaveTypeURL, nil)
+	if err != nil {
+		return leaveTypes, err
+	}
+	defer resp.Body.Close()
+
+	if err := json.NewDecoder(resp.Body).Decode(&leaveTypes); err != nil {
+		return leaveTypes, err
+	}
+
+	return leaveTypes, nil
+}
+
+// UpdateLeaveType update the given leave type id with the given
+// request data
+func (c Client) UpdateLeaveType(id string, lt UpdateLeaveTypeRequest) (LeaveType, error) {
+	var leaveType LeaveType
+
+	bytes, err := json.Marshal(lt)
+	if err != nil {
+		return leaveType, err
+	}
+
+	resp, err := c.put(leaveTypeURL+"/"+id, bytes)
+	if err != nil {
+		return leaveType, err
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&leaveType); err != nil {
+		return leaveType, err
+	}
+
+	return leaveType, nil
+}
+
+// Leave contains all the leave information
+type Leave struct {
+	ID          int    `json:"id"`
+	Description string `json:"description"`
+	EmployeeID  int    `json:"employee_id"`
+	FinishOn    string `json:"finish_on"`
+	HalfDay     string `json:"half_day"`
+	LeaveTypeID int    `json:"leave_type_id"`
+	StartOn     string `json:"start_on"`
+}
+
+// CreateLeaveRequest keeps the information needed
+// for create a new leave
+type CreateLeaveRequest struct {
+	Description string `json:"description,ompitempty"`
+	EmployeeID  int    `json:"employee_id"`
+	FinishOn    string `json:"finish_on"`
+	HalfDay     string `json:"half_day,ompitempty"`
+	LeaveTypeID int    `json:"leave_type_id"`
+	StartOn     string `json:"start_on"`
+}
+
+// UpdateLeaveRequest keeps the information needed
+// for update a leave
+type UpdateLeaveRequest struct {
+	Description string `json:"description,ompitempty"`
+	EmployeeID  int    `json:"employee_id,ompitempty"`
+	FinishOn    string `json:"finish_on,ompitempty"`
+	HalfDay     string `json:"half_day,ompitempty"`
+	LeaveTypeID int    `json:"leave_type_id,ompitempty"`
+	StartOn     string `json:"start_on,ompitempty"`
+}
+
+// CreateLeave creates a new leave.
+// Admins can create leaves for all employees,
+// regular users are restricted to themselves and employees they manage.
+func (c Client) CreateLeave(l CreateLeaveRequest) (Leave, error) {
+	var leave Leave
+
+	bytes, err := json.Marshal(l)
+	if err != nil {
+		return leave, err
+	}
+
+	resp, err := c.post(leaveURL, bytes)
+	if err != nil {
+		return leave, err
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&leave); err != nil {
+		return leave, err
+	}
+
+	return leave, nil
+}
+
+// DeleteLeave deletes an existing leave.
+// Admins can delete leaves for all employees,
+// regular users are restricted to themselves and employees they manage.
+// Restrictions apply if the leave has already started.
+func (c Client) DeleteLeave(id string) error {
+	_, err := c.delete(leaveURL + "/" + id)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ListLeaves gets all leaves from your company.
+func (c Client) ListLeaves() ([]Leave, error) {
```

Comment:
I saw this in other parts of the code as well, probably instead of using

```
var leaves []Leave
```

We can change the function and use 

```
func (c Client) ListLeaves() (leaves []Leave, err error)
```

What do you think?

Reply:
I prefer to maintain it as it is in order to improve the readability of our code

Reply:
cool